### PR TITLE
Add custom message in the case of a `invalid beatmap_hash`

### LINKED ISSUE #27781: Playing pending beatmap with an update available throws `invalid beatmap_hash` to user
### Type

Game behaviour

### Bug description

This type of notification should be hidden from user.
Maybe reword to "Your score will not be submitted. Please, update the beatmapset."

### Screenshots or videos

https://github.com/ppy/osu/assets/86934170/c810d6f6-6a31-45b2-a976-e51dd0f1cecf

### Version

2024.312.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14852945/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by 64ArthurAraujo**
Closes #27781, very small PR just adding a case for the "invalid beatmap_hash" `exception.Message`. Now it will show
a better message: "A new version of this beatmapset is available please update. Your score will not be submitted."

### CODE REVIEW

**Comment by SupDos** [comment]
You're missing a comma: `available, please`

Personally I think a better message for this would be
`This beatmap doesn't match the official version, please update or re-download this beatmap.\n\n{whatWillHappen}`

**Comment by bdach** [comment]
This message doesn't necessarily mean that an update to the beatmap is available. It might also mean that the beatmap has been modified (i.e. by changing it in stable and importing from there). Or it might mean that the beatmap was intentionally maliciously modified. Or it might mean corruption. And so on and so forth.

I'm not sure how to feel about this change in that light.

File: osu.Game/Screens/Play/SubmittingPlayer.cs [hunk 1]
```diff
@@ -152,6 +152,10 @@ void handleTokenFailure(Exception exception, bool displayNotification = false)
                                 Logger.Log($"Please ensure that you are using the latest version of the official game releases.\n\n{whatWillHappen}", level: LogLevel.Important);
                                 break;
 
+                            case @"invalid beatmap hash":
+                                Logger.Log($"A new version of this beatmapset is available please update. \n\n{whatWillHappen}", level: LogLevel.Important);
```
Comment (Walavouchey) [inline osu.Game/Screens/Play/SubmittingPlayer.cs:156]: slightly modified suggestion based on the above

```suggestion
                                Logger.Log($"This beatmap does not match the online version. Please update or redownload it.\n\n{whatWillHappen}", level: LogLevel.Important);
```

**Review by peppy — APPROVED** [review]

File: osu.Game/Screens/Play/SubmittingPlayer.cs (hunk 1 above)
Comment (Loreos7) [reply osu.Game/Screens/Play/SubmittingPlayer.cs]: You didn't include main purpose for an update. Should be:
"This beatmap does not match the online version. Please update or redownload it in order to submit a score.

Comment (SupDos) [reply osu.Game/Screens/Play/SubmittingPlayer.cs]: `{whatWillHappen}` has that covered with "Your score will not be submitted." - I don't think it's necessary to mention score submission twice here.
